alvr/graphics/src/stream.rs

```diff
@@ -304,89 +304,102 @@ impl StreamRenderer {
 fn set_passthrough_push_constants(render_pass: &mut RenderPass, config: Option<&PassthroughMode>) {
     const DEG_TO_NORM: f32 = 1. / 360.;
 
+    fn set_u32(render_pass: &mut RenderPass, offset: u32, value: u32) {
+        render_pass.set_push_constants(ShaderStages::VERTEX_FRAGMENT, offset, &value.to_le_bytes());
+    }
+
     fn set_float(render_pass: &mut RenderPass, offset: u32, value: f32) {
         render_pass.set_push_constants(ShaderStages::VERTEX_FRAGMENT, offset, &value.to_le_bytes());
     }
 
+    fn set_vec4(render_pass: &mut RenderPass, offset: u32, value: Vec4) {
+        render_pass.set_push_constants(
+            ShaderStages::VERTEX_FRAGMENT,
+            offset,
+            &value.x.to_le_bytes(),
+        );
+        render_pass.set_push_constants(
+            ShaderStages::VERTEX_FRAGMENT,
+            offset + FLOAT_SIZE,
+            &value.y.to_le_bytes(),
+        );
+        render_pass.set_push_constants(
+            ShaderStages::VERTEX_FRAGMENT,
+            offset + 2 * FLOAT_SIZE,
+            &value.z.to_le_bytes(),
+        );
+        render_pass.set_push_constants(
+            ShaderStages::VERTEX_FRAGMENT,
+            offset + 3 * FLOAT_SIZE,
+            &value.w.to_le_bytes(),
+        );
+    }
+
     match config {
-        Some(PassthroughMode::AugmentedReality { brightness }) => {
-            set_float(render_pass, ALPHA_CONST_OFFSET, 1. - brightness);
-            set_float(render_pass, ENABLE_CHROMA_KEY_CONST_OFFSET, 0.);
+        None => {
+            set_u32(render_pass, PASSTHROUGH_MODE_OFFSET, 0);
         }
-        Some(PassthroughMode::Blend { opacity }) => {
-            set_float(render_pass, ALPHA_CONST_OFFSET, 1. - opacity);
-            set_float(render_pass, ENABLE_CHROMA_KEY_CONST_OFFSET, 0.);
+        Some(PassthroughMode::Blend { threshold, .. }) => {
+            set_u32(render_pass, PASSTHROUGH_MODE_OFFSET, 1);
+            set_float(render_pass, ALPHA_CONST_OFFSET, 1. - threshold);
```

Comment:
I just noticed we could theoretically make the 'no passthrough' case be handled by just setting the alpha to 1. Maybe this would slightly speed up the shader by removing one conditional, tho since all shaders in one invocation take the same branch it's probably not by a whole lot.

Reply:
I think it being done separately is just a mental leftover from the case where this was handled by specialization constants instead of always dynamically.

Reply:
Literally the code was doing that before this PR, but I added the case to make it cleaner I guess

Reply:
The reason why I don't want to do it like you said is because it feels unclean. Each PASSTHROUGH_MODE has its own data associated with it. If we share the alpha parameter between no passthrough and blend modes, there is also nothing that shows evidently that alpha is not actually used for chroma keying. And in the end this doesn't change the side of push constants, it would just remove a if case in the shader but i really don't think it matters